source/manual/conventions-for-rails-applications.html.md

```diff
@@ -93,6 +93,44 @@ introduced by these gems.
 [slimmer]: https://github.com/alphagov/slimmer
 [issue-example]: https://github.com/alphagov/govuk_app_config/issues/121
 
+### Gemfile organisation
+
+Aim for your Gemfile to feel consistent, logical and concise. It is easy for
+these files to become confusing with arbitrary orderings and a sporadic
+approach to versioning ([example][confusing-gemfile]). However, this can
+be avoided by following simple conventions.
+
+The first gem in your Gemfile should be `"rails"` as the root dependency of
+the application. This should have the version of Rails specified as an absolute
+version number (for example `"6.0.3.4"` rather than relative `"~> 6.0"`) which
+stops any other gems being able to alter the version of Rails used.
+
+You should then declare the other gems your application needs to run in
+all environments, which should be followed by
+[groups](https://bundler.io/v1.5/groups.html) to specify gem dependencies
+for particular environments (typically development and test).
+
+You should avoid specifying the version of a gem (known as pinning) unless
+you have a specific need to do so. Pinning a version is rarely necessary
+as the Gemfile.lock file already stores the particular version used
+and typically we intend to keep applications compatible with a current
+version of a gem. Avoiding this makes future readers of your Gemfile not need
+to wonder if a choice was arbitrary or specific, and it makes the file an
+easier read - [example][email-alert-api-gemfile].
+
+Should you have a need to specify a particular version of the gem (for example,
+to indicate lack of compatibility with a newer release) leave a comment to
+explain the particular version. This documents why we need to care
+about the gem version. For example:
+
+```
+# We need a 6.x release to interface with Elasticsearch 6
+gem "elasticsearch", "~> 6"
```

Comment:
A very minor/picky comment, that I don't feel super strongly on but anyway...

```suggestion
gem "elasticsearch", "~> 6" # We need a 6.x release to interface with Elasticsearch 6
```


I've experienced times in the past where it's easy to miss a comment like this if it's not on the line itself. When Dependabot opens a PR for example, the diff highlights the line itself and the comment is very obvious, whereas if it's on the line above or below it's much easier to miss [(example diff)](https://github.com/alphagov/datagovuk_publish/pull/1013/files).

Reply:
Sure thanks, does seems better on the line if we can phrase it concisely enough but can't be a constant rule as we'd not want a single line for something. I'll see if I can amend the text to not be specific on placement.